### ISSUE TITLE
Add a backpropogation algorithm

### ISSUE THREAD

**Issue by MrGelatine** [mmcs-ruby/silicium#111]
Algorithm that used to update weights and other hyperparametrs of neural network based on loss function
